perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1167,9 +1167,11 @@ public void sendPasswordResetLinkEmail(PerunSession sess, Member member, String
 			throw new InternalErrorException(ex);
 		}
 
-		String mailAddress = (String) mailAttribute.getValue();
+		if (mailAttribute != null) {
```

Comment:
This is not a correct fix. The mailAttribute can be null if the given attributeDefinition is not `user` or `member` attribute. Instead, if the value is null, you should throw some Exception. I would suggest throwing InternalErrorException since we don't have an appropriate Exception for this use case.